LINQ to XML Object Loading step

jmarkman opened the pull request:
## Summary

- Added path construction step and loading step
- Added method syntax versions of queries
- Added links to the "Load XML from files or streams" line to corresponding docs pages 

Fixes #6256

jmarkman committed:
added object loading step
- added method syntax versions of queries
- added path construction step and loading step
- added link to How to: Load XML from a File (C#)

jmarkman committed:
moved link to more appropriate section and added stream link

BillWagner commented on docs/csharp/programming-guide/concepts/linq/linq-to-xml-overview.md:
````diff
@@ -24,26 +24,43 @@ XML has been widely adopted as a way to format data in many contexts. For exampl
-  
+```csharp
+// Load the XML file containing the purchase orders
+string folderPath = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
````
This won't work on .NET Core on Mac or Linux. We should update this to load a file from the current directory, or parse a constant string declared in the sample.

jmarkman replied:
I'm afraid I don't have the means to try this out on MacOS, but it looks like getting the file from the current directory will work for our crossplatform needs:
![chrome_2018-10-30_00-40-54](https://user-images.githubusercontent.com/8032019/47696316-9beba580-dbdc-11e8-85b5-fc7bebe458be.png)
I've done the majority of my development on a Windows machine for both work and leisure, so I haven't had much opportunity to play with .NET Core on other operating systems.

BillWagner replied:
I like the idea of using the current directory. That works great.

BillWagner commented on docs/csharp/programming-guide/concepts/linq/linq-to-xml-overview.md:
````diff
@@ -24,26 +24,43 @@ XML has been widely adopted as a way to format data in many contexts. For exampl
+This can be rewritten in method syntax form:
+
+```csharp
+IEnumerable<string> partNos = purchaseOrder.Descendants("Item").Select(x => x.Attribute("PartNumber"));
````
You should add a cast or `ToString` method so the type of the `Select` matches.

jmarkman replied:
An easy fix, I'll add it asap.

BillWagner commented on docs/csharp/programming-guide/concepts/linq/linq-to-xml-overview.md:
````diff
@@ -24,26 +24,43 @@ XML has been widely adopted as a way to format data in many contexts. For exampl
-select item;  
+```csharp 
+// Load the XML file containing the purchase orders
+string folderPath = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);
````
Same comment on the path here.

jmarkman replied:
Ditto for the response. I'll change this.

BillWagner reviewed (changes requested):
Overall, I like the additions you've made here @jmarkman It's a nice improvement.

I had a couple comments regarding type compatibility and loading from paths and files in a cross platform manner. Once we address those, this is ready to merge. 

Thanks for contributing to docs! 🎆

Commit:
updated based on feedback
- implemented platform-agnostic file loading code snippet
- added cast on the Select method at line 38
- updated ms.date

Commit:
i guess i didn't include the cast in the previous commit

BillWagner commented:
Thanks for making the updates @jmarkman I've approved the changes, and I'll :shipit: as soon as the build finishes.

You should see the changes on the live site in the next day or two, on our regular publishing cycle. Thanks again for contributing to docs! We really appreciate it. 🎆
